stablehlo/dialect/Version.h

```diff
@@ -40,6 +40,12 @@ class Version {
   /// Return a Version representing the current VHLO dialect version.
   static Version getCurrentVersion() { return Version(1, 7, 9); }
 
+  /// Return CompatibilityRequirement::WEEK_4 VHLO dialect version.
+  static Version getWeek4Version() { return Version(1, 7, 1); }
```

Comment:
Can we make these functions in the Version.cpp file, or maybe even variables within the function itself?

I don't want these methods getting explicitly called anywhere, all references should go via the WEEK_4/WEEK_12 enums, so want to avoid exposing these functions publicly / keep the header file interface clean.

Reply:
Otherwise all this LGTM, and once submitted, we'll build it into our oncall versioning rotation, so every PR which bumps version (i.e. https://github.com/openxla/stablehlo/pull/2610) will then also call this script.

Reply:
Changes:
- Removed getWeek4Version and getWeek12Version from Version.h
- Added comments such as `// WEEK_4 ANCHOR: DO NOT MODIFY` to `return Version(a, b, c);` in Version.cpp
- The scripts uses these anchors to update week_4 and week_12 versions in Version.cpp now
- Added validation that new values in Version.h and Version.cpp are present. sed exits with exit code 0 if replacements were not done.